### ISSUE TITLE
uninitialized constant Decanter::Extensions::ActiveRecord::Base

### ISSUE THREAD

**Issue by blimey85** [LaunchPadLab/decanter#43]
Just hit a new issue while setting up the Mailboxer gem in my project.
```
ActionView::Template::Error at /conversations/1
uninitialized constant Decanter::Extensions::ActiveRecord::Base
```
I'm not sure what info to provide on this one. It's not giving me anything specific to work with, or not that I understand anyway. I'm using the Mailboxer gem and with Decanter removed from my Gemfile, it works. With that in there, I get the above error.

My controller method:
```
  def show
    @conversation = current_user.mailbox.conversations.find(params[:id])
    @conversation.mark_as_read(current_user)
    @message = Mailboxer::Message.new
  end
```
Relevant section of the view:
```
h1= @conversation.subject
h4 Participants
- @conversation.participants.each do |participant|
  div= participant.fullname
```
It's the third line of the view that's causing it. I did some testing in the console and while it has no issue with @conversation, it's not happy with @convesation.participants with Decanter active. That should trigger a query that looks up the participants and it's a method provided by the Mailboxer gem.

Ideas on how I can troubleshoot this? Any additional info I can provide or testing I can do?

Rails 5.1.2, Ruby 2.4.1, Decanter 0.9.1, Mailboxer 0.15.1.

**Comment by francirp** [comment]
@blimey85 very peculiar indeed. I will take a look and get back to you ASAP

**Comment by blimey85** [comment]
If it would help, I can create a small sample app to illustrate the problem. Could make troubleshooting it much easier for you.

**Comment by francirp** [comment]
@blimey85 that would be very helpful - I appreciate the offer. Let me know when you have it up on Github and I can take it from there.

**Comment by blimey85** [comment]
Ok, I have a test app set up. It's embarrassingly ugly but serves its purpose. You should find it here: [https://github.com/blimey85/decanter_mailboxer_issue.git](https://github.com/blimey85/decanter_mailboxer_issue.git) I set it up using Ruby 2.4.0 but should work with whatever. It's set to Rails 5.1.2 and I don't know if the same issue exists on other versions. Also uses sqlite3 to keep this self-contained, however my main project is using MySQL and the issue exists there as well.

**Comment by francirp** [comment]
Got it. Will take a look now @blimey85

**Comment by francirp** [comment]
@blimey85 sorry for the massive delay here. I wasn't able to figure it out, have you made any progress on your end? I'll dig more into it this week.

**Comment by blimey85** [comment]
Sorry, I haven't had a chance to look into it any more. I've been working on other parts of my project that don't need Decanter so just have it disabled for now.

**Comment by blimey85** [comment]
@francirp I spent a couple hours tonight trying to figure out what's causing the issue. The odd thing to me is that Decanter works when you're actually using it. Whatever is getting loaded and called is working fine in that context. It's only outside of this that the problem presents itself. Also of note, I have to engines within my project and one works fine while the other doesn't. Mailboxer is a much more complicated setup however. Is Decanter overriding anything in ActiveRecord or just adding to it? It seems like when Mailboxer does certain queries it's hitting Decanter code but it's not fully loaded or something.

It's frustrating. I really like Decanter and spent quite a bit of time getting it setup on several models. That said, I've gotta get my app working so for now I'm going to just pull it out and go in a different direction and then hopefully come back to Decanter later.

**Comment by francirp** [comment]
Sorry to hear that @blimey85 - I'm digging into it today. Sorry for the delay I've been so busy lately.

Will update you by end of day.

**Comment by francirp** [comment]
@blimey85 `bundle update decanter` and you should be good to go

**Comment by blimey85** [comment]
@francirp This is awesome news! I just updated and tested in my simple test app I created to illustrate the problem and the problem is gone. I'll update in my main project and do more testing but it looks like this solves the issue. I'll also look at what you changed to see if I can understand what the problem was and the solution. Thank you so much for addressing this. Saves me a lot of time on having to try to rework code. Thank you!!!
